src/main/java/be/seeseemelk/mockbukkit/block/BlockMock.java

```diff
@@ -308,15 +308,13 @@ public int getBlockPower()
 	@Override
 	public boolean isEmpty()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return this.getType().equals(Material.AIR);
 	}
 
 	@Override
 	public boolean isLiquid()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return this.getType().equals(Material.LAVA) || this.getType().equals(Material.WATER);
```

Comment:
Also you forgot one liquid: `BUBBLE_COLUMN`, water above magma blocks has its own Material even though it looks and behaves like water.

Reply:
Wow, I didn't even know this liquid existed.